### PR TITLE
Ehcache

### PR DESCRIPTION
**Pull request by ledsoft**
Introduces Ehcache to improve the performance of resource and vocabulary list retrieval.

### CODE REVIEW

File: src/main/java/cz/cvut/kbss/termit/service/repository/ResourceRepositoryService.java [hunk 1]
```diff
@@ -70,6 +74,30 @@ public ResourceRepositoryService(Validator validator, ResourceDao resourceDao,
         return resourceDao;
     }
 
+    @Cacheable
+    @Override
+    public List<Resource> findAll() {
+        return super.findAll();
+    }
+
+    @CacheEvict(allEntries = true)
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/service/repository/ResourceRepositoryService.java:83]: why these methods evict all entries?

**Review by psiotwo — COMMENTED** [review]
see comment

File: src/main/java/cz/cvut/kbss/termit/service/repository/ResourceRepositoryService.java (hunk 1 above)
Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/service/repository/ResourceRepositoryService.java:83]: See javadoc to `CacheConfig` - there is actually only one entry in the cache and it contains the whole list of elements.

Comment (psiotwo) [reply src/main/java/cz/cvut/kbss/termit/service/repository/ResourceRepositoryService.java:83]: ok, makes sense.

**Review by psiotwo — APPROVED** [review]

**Pull request merged**